lib/utils/injectRefreshEntry.js

```diff
@@ -29,11 +30,23 @@ function isSocketEntry(entry) {
  * Injects an entry to the bundle for react-refresh.
  * @param {WebpackEntry} [originalEntry] A Webpack entry object.
  * @param {import('../types').NormalizedPluginOptions} options Configuration options for this plugin.
+ * @param {import('webpack').Compiler} compiler A webpack compiler object.
  * @returns {WebpackEntry} An injected entry object.
  */
-function injectRefreshEntry(originalEntry, options) {
+function injectRefreshEntry(originalEntry, options, compiler) {
   /** @type {Record<string, *>} */
   let resourceQuery = {};
+
+  if (compiler && compiler.options && compiler.options.devServer) {
+    resourceQuery.domain = createDomain(compiler.options);
+    if (compiler.options.devServer.sockHost)
+      resourceQuery.sockHost = compiler.options.devServer.sockHost;
+    if (compiler.options.devServer.sockPath)
+      resourceQuery.sockPath = compiler.options.devServer.sockPath;
+    if (compiler.options.devServer.sockPort)
+      resourceQuery.sockPath = compiler.options.devServer.sockPort;
+  }
+
   if (options.overlay) {
     options.overlay.sockHost && (resourceQuery.sockHost = options.overlay.sockHost);
```

Comment:
Shameless plug that solves my problem: Could we also add this here?

```suggestion
    options.overlay.sockProtocol && (resourceQuery.sockProtocol = options.overlay.sockProtocol);
    options.overlay.sockHost && (resourceQuery.sockHost = options.overlay.sockHost);
```

Reply:
I'm open to adding it if @pmmmwh wants it

Reply:
Turns out it’s not that trivial to add it. Opened a PR for it https://github.com/pmmmwh/react-refresh-webpack-plugin/pull/242